# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1536309919 -7200
#      Fri Sep 07 10:45:19 2018 +0200
# Branch stable
# Node ID 9578af67a0d43ee177b559e023687378d695f9ef
# Parent  1448204783788c16cd55d7b9a39b9a292d144b3f
Document that image x/ydata indicate the centers of corner pixels (bug #54610).

* image.m, imagesc.m, imshow.m: Rephrase doc for x and y arguments to make it
   clear that they indicate the coordinates of the center of pixels.

* genpropdoc.m: Ditto.

diff --git a/doc/interpreter/genpropdoc.m b/doc/interpreter/genpropdoc.m
--- a/doc/interpreter/genpropdoc.m
+++ b/doc/interpreter/genpropdoc.m
@@ -1143,25 +1143,25 @@ measured in degrees.";
 
       case "cdatamapping"
       case "displayname"
         s.doc = "Text for the legend entry corresponding to this image.";
         s.valid = valid_cellstring;
 
       case "xdata"
         s.doc = "Two-element vector @code{[xmin xmax]} specifying the x \
-coordinates of the first and last columns of the image.\n\
+coordinates of the centers of the first and last columns of the image.\n\
 \n\
 Setting @code{xdata} to the empty matrix ([]) will restore the default value \
 of @code{[1 columns(image)]}.";
         s.valid = valid_2elvec;
 
       case "ydata"
         s.doc = "Two-element vector @code{[ymin ymax]} specifying the y \
-coordinates of the first and last rows of the image.\n\
+coordinates of the centers of the first and last rows of the image.\n\
 \n\
 Setting @code{ydata} to the empty matrix ([]) will restore the default value \
 of @code{[1 rows(image)]}.";
         s.valid = valid_2elvec;
 
     endswitch
 
   ## Surface properties
diff --git a/scripts/image/image.m b/scripts/image/image.m
--- a/scripts/image/image.m
+++ b/scripts/image/image.m
@@ -22,21 +22,22 @@
 ## @deftypefnx {} {} image (@dots{}, "@var{prop}", @var{val}, @dots{})
 ## @deftypefnx {} {} image ("@var{prop1}", @var{val1}, @dots{})
 ## @deftypefnx {} {@var{h} =} image (@dots{})
 ## Display a matrix as an indexed color image.
 ##
 ## The elements of @var{img} are indices into the current colormap.
 ##
 ## @var{x} and @var{y} are optional 2-element vectors, @w{@code{[min, max]}},
-## which specify the range for the axis labels.  If a range is specified as
-## @w{@code{[max, min]}} then the image will be reversed along that axis.  For
-## convenience, @var{x} and @var{y} may be specified as N-element vectors
-## matching the length of the data in @var{img}.  However, only the first and
-## last elements will be used to determine the axis limits.
+## which specify the coordinates of the centers of the corner pixels.
+## If a range is specified as @w{@code{[max, min]}} then the image will be
+## reversed along that axis.  For convenience, @var{x} and @var{y} may be
+## specified as N-element vectors matching the length of the data in @var{img}.
+## However, only the first and last elements will be used to determine the axis
+## limits.
 ##
 ## Multiple property/value pairs may be specified for the image object, but
 ## they must appear in pairs.
 ##
 ## The optional return value @var{h} is a graphics handle to the image.
 ##
 ## Implementation Note: The origin (0, 0) for images is located in the
 ## upper left.  For ordinary plots, the origin is located in the lower
diff --git a/scripts/image/imagesc.m b/scripts/image/imagesc.m
--- a/scripts/image/imagesc.m
+++ b/scripts/image/imagesc.m
@@ -25,20 +25,23 @@
 ## @deftypefnx {} {} imagesc (@var{hax}, @dots{})
 ## @deftypefnx {} {@var{h} =} imagesc (@dots{})
 ## Display a scaled version of the matrix @var{img} as a color image.
 ##
 ## The colormap is scaled so that the entries of the matrix occupy the entire
 ## colormap.  If @code{@var{climits} = [@var{lo}, @var{hi}]} is given, then
 ## that range is set to the @qcode{"clim"} of the current axes.
 ##
-## The axis values corresponding to the matrix elements are specified in
-## @var{x} and @var{y}, either as pairs giving the minimum and maximum
-## values for the respective axes, or as values for each row and column
-## of the matrix @var{img}.
+## @var{x} and @var{y} are optional 2-element vectors, @w{@code{[min, max]}},
+## which specify the coordinates of the centers of the corner pixels.
+## If a range is specified as @w{@code{[max, min]}} then the image will be
+## reversed along that axis.  For convenience, @var{x} and @var{y} may be
+## specified as N-element vectors matching the length of the data in @var{img}.
+## However, only the first and last elements will be used to determine the axis
+## limits.
 ##
 ## The optional return value @var{h} is a graphics handle to the image.
 ##
 ## Calling Forms: The @code{imagesc} function can be called in two forms:
 ## High-Level and Low-Level.  When invoked with normal options, the High-Level
 ## form is used which first calls @code{newplot} to prepare the graphic figure
 ## and axes.  When the only inputs to @code{image} are property/value pairs
 ## the Low-Level form is used which creates a new instance of an image object
diff --git a/scripts/image/imshow.m b/scripts/image/imshow.m
--- a/scripts/image/imshow.m
+++ b/scripts/image/imshow.m
@@ -43,25 +43,27 @@
 ## @table @asis
 ## @item @qcode{"displayrange"}
 ## @var{value1} is the display range as described above.
 ##
 ## @item @qcode{"colormap"}
 ## @var{value1} is the colormap to use when displaying an indexed image.
 ##
 ## @item @qcode{"xdata"}
-## If @var{value1} is a two element vector, it must contain horizontal axis
-## limits in the form [xmin xmax]; Otherwise @var{value1} must be a vector and
-## only the first and last elements will be used for xmin and xmax
+## If @var{value1} is a 2-element vector, it must contain horizontal image
+## limits in the form [xmin, xmax], where xmin and xmax are the abscissa of
+## the centers of the corner pixels.  Otherwise @var{value1} must be a vector
+## and only the first and last elements will be used for xmin and xmax
 ## respectively.
 ##
 ## @item @qcode{"ydata"}
-## If @var{value1} is a two element vector, it must contain vertical axis
-## limits in the form [ymin ymax]; Otherwise @var{value1} must be a vector and
-## only the first and last elements will be used for ymin and ymax
+## If @var{value1} is a 2-element vector, it must contain vertical image
+## limits in the form [ymin, ymax], where ymin and ymax are the ordinates of
+## the center of the corner pixels.  Otherwise @var{value1} must be a vector
+## and only the first and last elements will be used for ymin and ymax
 ## respectively.
 ##
 ## @end table
 ##
 ## The optional return value @var{h} is a graphics handle to the image.
 ## @seealso{image, imagesc, colormap, gray2ind, rgb2ind}
 ## @end deftypefn
 
